pkg/proxy/winkernel/hcnutils.go

```diff
@@ -29,15 +29,54 @@ import (
 	"strings"
 )
 
-type hnsV2 struct{}
+type HCN interface {
+	GetNetworkByName(networkName string) (*hcn.HostComputeNetwork, error)
+	ListEndpointsOfNetwork(networkId string) ([]hcn.HostComputeEndpoint, error)
+	GetEndpointByID(endpointId string) (*hcn.HostComputeEndpoint, error)
+	ListEndpoints() ([]hcn.HostComputeEndpoint, error)
+	GetEndpointByName(endpointName string) (*hcn.HostComputeEndpoint, error)
+	ListLoadBalancers() ([]hcn.HostComputeLoadBalancer, error)
+	GetLoadBalancerByID(loadBalancerId string) (*hcn.HostComputeLoadBalancer, error)
+	CreateEndpoint(endpoint *hcn.HostComputeEndpoint, network *hcn.HostComputeNetwork) (*hcn.HostComputeEndpoint, error)
+	CreateLoadBalancer(loadbalancer *hcn.HostComputeLoadBalancer) (*hcn.HostComputeLoadBalancer, error)
+	CreateRemoteEndpoint(endpoint *hcn.HostComputeEndpoint, network *hcn.HostComputeNetwork) (*hcn.HostComputeEndpoint, error)
+	DeleteLoadBalancer(loadbalancer *hcn.HostComputeLoadBalancer) error
+	DeleteEndpoint(endpoint *hcn.HostComputeEndpoint) error
+}
+
+// The ihcn structure represents the real hcn, so we can call the real hcn.methods whenever we need them.
+// If you want to use hcnutils methods on the real hcn you should create it this way: hcnutils := NewHCNUtils(&ihcn{})
+// Otherwise, if you want to use it on the fakeHCN (for testing purposes), use it this way: hcnutils := NewHCNUtils(&hcntesting.FakeHCN{})
+type ihcn struct{}
+
+type hcnutils struct {
+	hcninstance HCN
+}
+
+func NewHCNUtils(hcnImpl HCN) *hcnutils {
+	return &hcnutils{hcnImpl}
+}
+
+type HCNUtils interface {
```

Comment:
What is this interface? It looks similar to HCN interface and I don't see it being used anywhere. Am I missing something?

Reply:
Ok I think I understand this now. HCNUtils is the interface and hcnutils is the implementation of the HCNUtils interface. Can we use a uniform naming scheme for all the interfaces? Or else it is confusing.